pkgs/applications/science/math/sage/sagedoc.nix

```diff
@@ -24,15 +29,25 @@ stdenv.mkDerivation rec {
     export HOME="$TMPDIR/sage_home"
     mkdir -p "$HOME"
 
+    # run bootstrap script to generate Sage spkg docs, because unfortunately some unrelated doc
+    # pages link to them. it needs a few ugly (but self-contained) hacks for a standalone run.
+    cp -r "${src}/build" "$HOME"
+    chmod -R 755 "$HOME/build"
+    sed -i "/assert/d" "$HOME/build/sage_bootstrap/env.py"
+    sed -i "s|sage-bootstrap-python|python3|" "$HOME/build/bin/sage-package"
+    patchShebangs "$HOME/build/bin/sage-package"
+    pushd "$SAGE_DOC_SRC_OVERRIDE"
+    sed -i "s|OUTPUT_DIR=\"src/doc/|OUTPUT_DIR=\"$SAGE_DOC_SRC_OVERRIDE/|" bootstrap
+    PATH="$HOME/build/bin:$PATH" SAGE_ROOT="${src}" ./bootstrap
+    popd
```

Comment:
These hacky lines are the practical downside to running the bootstrap script. I kept them contained to `sagedoc.nix`.